src/cpp/desktop/DesktopMainWindow.hpp

```diff
@@ -40,7 +40,11 @@ class MainWindow : public GwtWindow
 public:
    explicit MainWindow(QUrl url=QUrl());
 
+private:
+   static MainWindow *instance;
+
 public:
+   static MainWindow *getInstance();
```

Comment:
I'm not a fan of this. Why not have `ScriptCommand.mm` accept a `MainWindow` as part of its constructor? Or if that's not possible for some reason, have a method that it can call to 'bind' to a window instance.

Reply:
I am not sure if it is possible to dynamically initialize `NSScriptCommand`, akaik, it is usually initialized automatically by the application. This implementation just mirrors `desktop-mac`'s: https://github.com/rstudio/rstudio/blob/54caf1f66b724680091623b6eca157870e3a3e6c/src/cpp/desktop-mac/MainFrameController.mm#L37

Reply:
Perhaps the `evaluateRScriptCommand` class could gain a function e.g. `- initialize(MainWindow* window)` that gets called by the `MainWindow` class when appropriate? And that could take care of setting the `MainWindow` instance so that it can be used.